query-engine/query-engine-node-api/src/node_drivers/engine.rs

```diff
@@ -0,0 +1,140 @@
+use crate::engine::QueryEngine;
+
+#[allow(unused_imports)]
+use napi::bindgen_prelude::Promise as JsPromise;
+
+use napi::{Env, JsFunction, JsObject, JsString, JsUnknown};
+use napi_derive::napi;
+
+pub struct NodejsFunctionContext {
+    env: Env,
+    pub query_raw: JsFunction,
+    pub query_raw_typed: JsFunction,
+    pub execute_raw: JsFunction,
+    pub execute_raw_typed: JsFunction,
+    pub version: JsFunction,
+}
+
+impl NodejsFunctionContext {
+    pub fn query_raw(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("SELECT * FROM users").unwrap();
+        let _promise: JsUnknown = self.query_raw.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise
```

Comment:
I have no clue how to convert `napi::JsUnknown` into `napi::bindgen_prelude::Promise`.

Reply:
I went with a similar route using `ThreadsafeFunction<T>`s